src/weights.jl

```diff
@@ -2,44 +2,181 @@
 ###### Weight vector #####
 
 if VERSION < v"0.6.0-dev.2123"
-    immutable WeightVec{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
-        values::V
-        sum::S
-    end
+    abstract AbstractWeights{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
 else
-    immutable WeightVec{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T}
-        values::V
-        sum::S
+    abstract AbstractWeights{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T}
+end
+
+"""
+    `@weights name`
+
+Generates a new generic weight type with specified `name`, which subtypes `AbstractWeights`
+and stores the `values` (`V<:RealVector`) and `sum` (`S<:Real`).
+"""
+macro weights(name)
+    return quote
+        if VERSION < v"0.6.0-dev.2123"
+            immutable $name{S<:Real, T<:Real, V<:RealVector} <: AbstractWeights{S, T, V}
+                values::V
+                sum::S
+            end
+        else
+            immutable $name{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractWeights{S, T, V}
+                values::V
+                sum::S
+            end
+        end
     end
 end
 
+eltype(wv::AbstractWeights) = eltype(wv.values)
+length(wv::AbstractWeights) = length(wv.values)
+values(wv::AbstractWeights) = wv.values
+sum(wv::AbstractWeights) = wv.sum
+isempty(wv::AbstractWeights) = isempty(wv.values)
+
+Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
+Base.size(wv::AbstractWeights) = size(wv.values)
+
 """
-    WeightVec(vs, [wsum])
+    bias(n::Integer, [corrected])
 
-Construct a `WeightVec` with weight values `vs` and sum of weights `wsum`.
+Computes the corrected (default) or uncorrected bias for any `n` observations.
+
+```math
+\fraction{1}{n - 1}
+```
+"""
+bias(n::Integer, corrected=true) = inv(n - Int(corrected))
+
+"""
+    bias(w::AbstractWeights, [corrected])
+
+Computes the corrected (default) or uncorrected bias for any weight vector.
+The default equation assumes analytic/precision/reliability weights and determines the
+bias as:
+
+```math
+\fraction{1}{∑w × (1 - ∑(w'²))}
+```
+where w' represents the normalized weights
+"""
+function bias(w::AbstractWeights, corrected=true)
+    s = sum(w)
+    if corrected
+        return inv(s * (1 - sum(normalize(values(w), 1) .^ 2)))
```

Comment:
`normalize` and `.^2` are going to create copies of the weights, which should be avoided. Formula should be adapted so that no allocation happens at all. Maybe you'll need a loop for that.